nexus/db-queries/tests/output/reserve_external_ip_pool.sql

```diff
@@ -0,0 +1,16 @@
+UPDATE
+  ip_pool
+SET
+  reservation_type = $1, time_modified = now()
+WHERE
+  id = $2
+  AND time_deleted IS NULL
+  AND reservation_type = $3
+  AND CAST(
+      IF(
+        EXISTS(SELECT 1 FROM ip_pool_resource WHERE ip_pool_id = $4 LIMIT 1),
```

Comment:
More thinking out loud - this all looks correct to me, but I'm vaguely worried that our checks to switch between reservation types are so different. If I want to change something away from "for external silos", I have to check the `ip_pool_resource` table, but if I want to change something away from "for Oxide internal", I have to check the `external_ip` table. Should rows in `external_ip` also have corresponding rows in `ip_pool_resource`?